### PR TITLE
Cross language compatibility: case insensitivity flag

### PR DESCRIPTION
**Pull request by TomSellers**
## Description
This PR moves the inline flag `(?i)` (case insensitivity)  to the beginning of the regex. In Python this flags has global meaning and use of it anywhere other than the start of the pattern is deprecated. This does not change the function of this flag in current patterns for other languages.

See previous work in https://github.com/rapid7/recog/pull/367

This was caught by internal testing in `recog-go` and the fixes are required in order for us to implement and test the changes from the following PRs into `recog-go`:
- https://github.com/rapid7/recog-ruby/pull/12
- https://github.com/rapid7/recog/pull/547

## How Has This Been Tested?
`bundle exec rake tests`
`ruby bin/recog_verify xml/*.xml | grep -v WARN | grep -v SUMMARY`

## Types of changes>
- Content

## Checklist:

- [ ] I have updated the documentation accordingly (or changes are not required).
- [x] I have added tests to cover my changes (or new tests are not required).
- [ ] All new and existing tests passed.

### CODE REVIEW

File: xml/imap_banners.xml [hunk 1]
```diff
@@ -142,7 +142,6 @@
     <example>Dovecot (Debian) ready.</example>
     <example>[CAPABILITY IMAP4rev1 SASL-IR LOGIN-REFERRALS ID ENABLE IDLE LITERAL+ AUTH=PLAIN AUTH=LOGIN] Dovecot (Debian) ready.</example>
     <param pos="0" name="service.vendor" value="Dovecot"/>
-    <param pos="0" name="service.vendor" value="Dovecot"/>
```
Comment (TomSellers) [inline xml/imap_banners.xml:145]: Not flag related, but duplicate `param` use detected by tests in `recog-go`.

**Comment by mkienow-r7** [comment]
> This was caught by internal testing in recog-go ...

The hope behind the [Go Verify workflow](https://github.com/rapid7/recog/blob/c25eda8b5ec4e8b04311f41b66760222c43a27a8/.github/workflows/verify.yml#L70-L89) was to detect issues between the various recog language implementation. What sort of internal testing discovered this issue that was not discoverable when verifying the fingerprints against their examples via [cmd/recog_verify/main.go](https://github.com/runZeroInc/recog-go/blob/main/cmd/recog_verify/main.go)?

File: xml/dhcp_vendor_class.xml [hunk 2]
```diff
@@ -8,7 +8,7 @@
     and inform (8) messages.
   -->
 
-  <fingerprint pattern="^[Mm]fg=(?:Fuji)?(?i:Xerox);[Tt]yp=(?:MFP|AIO|[Pp]rinter);[Mm]od=(?:Xerox )?(\S+) ([a-zA-Z0-9]+).*;[Ss]er=([A-Z0-9]{9,10})(?:;[Ll]oc=.*)?$">
+  <fingerprint pattern="(?i)^[Mm]fg=(?:Fuji)?Xerox;[Tt]yp=(?:MFP|AIO|[Pp]rinter);[Mm]od=(?:Xerox )?(\S+) ([a-zA-Z0-9]+).*;[Ss]er=([A-Z0-9]{9,10})(?:;[Ll]oc=.*)?$">
```
Comment (mkienow-r7) [inline xml/dhcp_vendor_class.xml:11]: With case insensitivity applied to the whole pattern the explicit case matching is no longer necessary.
* `[Mm]`
* `[Tt]`
* `[Pp]`
* `[Ss]`
* `[Ll]`

Comment (TomSellers) [reply xml/dhcp_vendor_class.xml]: Good catch, fixed.

**Comment by TomSellers** [comment]
@mkienow-r7 

> The hope behind the [Go Verify workflow](https://github.com/rapid7/recog/blob/c25eda8b5ec4e8b04311f41b66760222c43a27a8/.github/workflows/verify.yml#L70-L89) was to detect issues between the various recog language implementation. What sort of internal testing discovered this issue that was not discoverable when verifying the fingerprints against their examples via [cmd/recog_verify/main.go](https://github.com/runZeroInc/recog-go/blob/main/cmd/recog_verify/main.go)?

Testing in the Go version of `recog_verify` should be the same as the ruby version so that we can test consistently.

The test for this condition is in `fingerprints_test.go` which would similar to the `spec` tests.

https://github.com/runZeroInc/recog-go/blob/1f868cde9c4b2e110f99d1114d1df4e530842f90/fingerprints_test.go#L51-L58

**Review by mkienow-r7 — APPROVED** [review]
Thank you for the contribution @TomSellers!

**Pull request merged**